samcli/lib/build/app_builder.py

```diff
@@ -0,0 +1,137 @@
+"""
+Builds the application
+"""
+
+import os
+import shutil
+from aws_lambda_builders.builder import LambdaBuilder
+
+
+class ApplicationBuilder(object):
+    """
+    Class to build an entire application. Currently, this class builds Lambda functions only, but there is nothing that
+    is stopping this class from supporting other resource types. Building in context of Lambda functions refer to
+    converting source code into artifacts that can be run on AWS Lambda
+    """
+
+    _builder_capabilities = {
+        "python2.7": {
+            "language": "python",
```

Comment:
Why not make this content a namedtuple instead?

Reply:
No reason. Dict was the quickest. I am changing this to something more general.